File: tests/scripts/DESeq2/DESeq2.py
import pandas as pd

from r_wrapper import base, stats, DESeq2


# read data
df_cts = pd.read_csv('count_data.csv', index_col=0)
df_coldata = pd.read_csv('col_data.csv', index_col=0)

# do DEA
dds = DESeq2.DESeqDataSetFromMatrix(
    countData=df_cts,
    colData=df_coldata,
    design=stats.as_formula('~ condition'))

dds = DESeq2.DESeq(dds)

res = DESeq2.results(dds)

# save result
df_res = base.as_data_frame(res)

(df_res.reset_index()
       .rename(columns={'index': 'gene'})
       .to_csv('result.csv', index=False))

File: tests/scripts/DESeq2/DESeq2.R
library("DESeq2")


df_cts <- read.csv("count_data.csv", row.names = 1)
df_coldata <- read.csv("col_data.csv", row.names = 1)

dds <- DESeqDataSetFromMatrix(
    countData = df_cts,
    colData = df_coldata,
    design = ~ condition
)

dds <- DESeq(dds)
res <- results(dds)

df_res <- as.data.frame(res)
df_res$gene <- rownames(df_res)
write.table(df_res, "result.csv", quote = FALSE, sep = ",", row.names = FALSE)
